nengo_spinnaker/simulator.py

```diff
@@ -341,6 +371,10 @@ def close(self):
             self.io_controller.close()
             self.controller.send_signal("stop")
 
+            # Destroy the job if we allocated one
+            if self.job is not None:
+                self.job.destroy()
```

Comment:
It will work for all cases that [`atexit`](https://docs.python.org/3.5/library/atexit.html) does; in my experience it is *always* called.